### PR TITLE
Add test for merging a fragment with a depth of 4

### PR DESCRIPTION
**Pull request by AnthonyMDev**
PR #562 highlighted that we are not testing fragment field merging past a depth of 3 entities. While we cannot test for every depth to infinity, this test confirms merging works to a depth of 4 and proves the edge case of that PR.

### CODE REVIEW

File: Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift [hunk 1]
```diff
@@ -4090,6 +4090,151 @@ class IRRootFieldBuilderTests: XCTestCase {
...
+          }
+        }
+      }
+    }    
+    """
+
+    // when
+    try await buildSubjectRootField()
+
+    let Object_Root = try XCTUnwrap(schema[object: "Root"])
```
Comment (calvincestari) [inline Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift:4155]: There is a warning that this variable is never used - is that intentional or highlighting something missed in the test?

**Review by calvincestari — COMMENTED** [review]
One question about the var assignment warning.

File: Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift (hunk 1 above)
Comment (AnthonyMDev) [reply Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift]: Oh you're right. For some reason the warning did not show up for me in Xcode? Fixed!

**Review by calvincestari — APPROVED** [review]
